autoload/maktaba/syscall.vim

```diff
@@ -455,24 +437,21 @@ endfunction
 
 ""
 " @private
-" Executes the asynchronous callback setup by @function(Syscall.CallAsync).
-" The callback must be of prototype: callback(result_dict) or legacy prototype
-" callback(env_dict, result_dict). The latter will be deprecated and stop
-" working in future versions of maktaba.
-function! maktaba#syscall#AsyncDone(stdout_file, stderr_file, exit_code)
+" Marks the SyscallInvocation associated with {invocation_id} finished with
+" given {exit_code} and executes its callback.
+function! maktaba#syscall#AsyncDone(invocation_id, exit_code)
   try
-    let l:callback_info = s:callbacks[a:stdout_file]
-    let l:return_data = {}
-    let l:return_data.status = a:exit_code
-    let l:return_data.stdout = join(readfile(a:stdout_file), "\n")
-    if filereadable(a:stderr_file)
-      let l:return_data.stderr = join(readfile(a:stderr_file), "\n")
-      call delete(a:stderr_file)
+    let l:invocation = s:pending_invocations[a:invocation_id]
```

Comment:
Is it worth trapping an unknown invocation ID explicitly?

(Perhaps not, since you'll catch E716 on L455 anyway.)